src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -174,13 +179,48 @@ const UsersFragment = ({
 
     const isUnfollowed = item.user ? !item.user.following : !item.following;
 
+    const handleOpenProfile = async (item) => {
+      if (profile?.myProfile?.user_id === item?.user_id) {
+        return null;
+      }
+
+      return navigation.push('OtherProfile', {
+        data: {
+          user_id: profile.myProfile.user_id,
+          other_id: item?.user_id || item?.userId,
+          username: item?.user?.name || item?.user?.username || item.username
+        }
+      });
+    };
+
+    const checkUserIsBlockHandle = async () => {
+      try {
+        setLoadingDM(true);
+        const sendData = {
+          user_id: item?.user_id || item?.userId
+        };
+        const members = [];
+        members.push(profile?.myProfile?.user_id, item?.user_id || item?.userId);
+        const processGetBlock = await checkUserBlock(sendData);
+        if (!processGetBlock.data.data.blocked && !processGetBlock.data.data.blocker) {
+          return createSignChat(members, item);
+        }
+        return handleOpenProfile(item);
+      } catch (e) {
+        console.log('error:', e);
+      }
+    };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleOpenProfile` and `checkUserIsBlockHandle` functions are defined inside another function which is not good for performance because they will be redefined every time the parent function runs. Consider moving these functions outside or use `useCallback` hook to prevent unnecessary redefinitions.

Also, error handling in `checkUserIsBlockHandle` could be improved. Currently, it just logs the error to the console. It would be better to show some feedback to the user or handle the error in a way that doesn't break the user experience.
```diff
- const handleOpenProfile = async (item) => {...};
- const checkUserIsBlockHandle = async () => {...};
+ const handleOpenProfile = useCallback(async (item) => {...}, []);
+ const checkUserIsBlockHandle = useCallback(async () => {...}, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->